platforms/avr/drivers/i2c_master.c

```diff
@@ -202,6 +202,22 @@ i2c_status_t i2c_writeReg(uint8_t devaddr, uint8_t regaddr, const uint8_t* data,
     return status;
 }
 
+i2c_status_t i2c_writeReg16(uint8_t devaddr, uint16_t regaddr, const uint8_t* data, uint16_t length, uint16_t timeout) {
+    i2c_status_t status = i2c_start(devaddr | 0x00, timeout);
+    if (status >= 0) {
+        i2c_write(regaddr >> 8, timeout);
```

Comment:
Looks like you're ignoring the return value here? I think you shouldn't ignore it.

Reply:
There's not much point without rewriting the function, as it would just be overwritten by the next `i2c_write()`, as that is for each data byte too. Apparently the only status that really matters with the current code is the last byte, as that's the one that will be returned.

Reply:
Umm, no, see the condition expression of the for loop below this piece of code. It checks status for every single byte, and ends the for loop early, if status is negative.
It's technically permissible for the device to nack the first register address byte, so this condition must be handled too.
There should be another nested if statement here. I was almost gonna say you can return if it's negative, but that's not good enough cause i2c_stop needs to be executed in case of failure too.

Reply:
My main point still stands that this first `i2c_write()`'s status would just be trashed by the second.

Reply:
I suppose that the idea is that the whole operation should be aborted if there is an error at any step, but the current version of the code does not do this and just ignores some errors.

Reply:
> My main point still stands that this first i2c_write()'s status would just be trashed by the second.

That's not true for the 8-bit version of the function. And my point is exactly that it shouldn't be true for the 16-bit version either.

Reply:
Of course it's not true for the 8-bit version? There's only one `i2c_write()` for the register address needed...

Reply:
Your latest commit "Check status of first i2c_write() in AVR writeReg16()" is exactly what I was aiming towards when Insaid 'nested if'.